### PR TITLE
ipn/localapi: define a cert dir for Synology DSM6

### LINKED ISSUE #4060: Failed to get cert on Synology
### What is the issue?

My win10 device can get the cert file successfully, but the Synology failed...
Running the command on Synology:
```
root@myNas:~# tailscale cert                                                                                                                                    
Usage: tailscale cert [flags] <domain>
For domain, use "mynas.****.ts.net".

root@myNas:~# tailscale cert mynas.****.ts.net
unexpected output: no delimiter

```

### Steps to reproduce

Run `tailscale cert <domain>` on a Synology device.  
Then it returns
 `unexpected output: no delimiter`

### Are there any recent changes that introduced the issue?

_No response_

### OS

Synology

### OS version

DSM6

### Tailscale version

1.22.0

### Bug report

BUG-4107f35d6ef47f772854e72ba476b9a07037c6ddfc410801bf5c252697827bcb-20220303170343Z-096b0250aa682d28

### PR DESCRIPTION
**Pull request by bradfitz**
Fixes #4060

### CODE REVIEW

File: version/distro/distro.go [hunk 1]
```diff
@@ -94,3 +95,23 @@ func freebsdDistro() Distro {
...
+// If not Synology, it reports 0.
+func DSMVersion() int {
+	if runtime.GOOS != "linux" {
+		return 0
+	}
+	if Get() != Synology {
+		return 0
+	}
+	v, _ := os.ReadFile("/etc/VERSION")
+	if bytes.Contains(v, []byte(`majorversion="6"`)) {
```
Comment (DentonGentry) [inline version/distro/distro.go:110]: There was at least one person who managed to get DSM5.1 sortof working. I think they had to disable something about certificate checking, as the CA roots were so old.

**Review by DentonGentry — APPROVED** [review]

File: version/distro/distro.go (hunk 1 above)
Comment (bradfitz) [reply version/distro/distro.go:110]: I don't want to start supporting DSM5. We never have, we don't have test devices, we rarely hear about it, etc.

File: version/distro/distro.go [hunk 2]
```diff
@@ -94,3 +95,23 @@ func freebsdDistro() Distro {
 	}
 	return ""
 }
+
+// DSMVersion reports the Synology DSM major version.
+//
+// If not Synology, it reports 0.
+func DSMVersion() int {
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply version/distro/distro.go:102]: But in the build code, I thought, in a different repo?

[the remark this answers is not in the document]
Comment (maisem) [reply version/distro/distro.go:102]: Oh I guess this is what I was referring to
```
func useAmbientCaps() bool {
	if distro.Get() != distro.Synology {
		return false
	}
	v, err := strconv.Atoi(os.Getenv("SYNOPKG_DSM_VERSION_MAJOR"))
	if err != nil {
		return false
	}
	return v >= 7
}
```

**Comment by bradfitz** [comment]
Works!  Needs DSM6 Update 3 or higher.

```
root@ds21bi:~# tailscale cert ds21bi.fitzpat.ts.net
Wrote public cert to ds21bi.fitzpat.ts.net.crt
Wrote private key to ds21bi.fitzpat.ts.net.key
```

File: version/distro/distro.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bradfitz) [reply version/distro/distro.go:102]: done

**Review by maisem — APPROVED** [review]

**Pull request merged**